internal/plugin6/grpc_provider.go

```diff
@@ -748,6 +788,103 @@ func (p *GRPCProvider) ReadDataSource(ctx context.Context, r providers.ReadDataS
 	return resp
 }
 
+func (p *GRPCProvider) OpenEphemeralResource(ctx context.Context, r providers.OpenEphemeralResourceRequest) (resp providers.OpenEphemeralResourceResponse) {
+	logger.Trace("GRPCProvider.v6: OpenEphemeralResource")
+
+	schema := p.GetProviderSchema(ctx)
+	if schema.Diagnostics.HasErrors() {
+		resp.Diagnostics = schema.Diagnostics
+		return resp
+	}
+
+	ephemeralResourceSchema, ok := schema.EphemeralResources[r.TypeName]
+	if !ok {
+		schema.Diagnostics = schema.Diagnostics.Append(fmt.Errorf("unknown ephemeral resource %q", r.TypeName))
```

Comment:
Why not set resp.Diagnostics = ... instead of temporarily storing the result in schema.Diagnostics?

Reply:
Good point. Reworked this to have better diagnostics collection.
Changes are in fb86ede8275ce6edc116778b829cc9abd08cf6a4